iceoryx_utils/source/cxx/timer.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_utils/cxx/timer.hpp"
+
+namespace iox
+{
+namespace cxx
+{
+Timer::Timer(const iox::units::Duration interval, const iox::units::Duration delayThreshold) noexcept
+    : m_interval(interval)
+    , m_delayThreshold(delayThreshold)
+{
+    start();
+}
+
+void Timer::start() noexcept
+{
+    stop();
+    auto waitResult = m_waitSemaphore.timedWait(m_interval, true);
+    cxx::Expects(!waitResult.has_error());
+    m_timeForNextActivation = now() + m_interval;
+}
+
+void Timer::start(const iox::units::Duration interval) noexcept
+{
+    m_interval = interval;
+    start();
+}
+
+void Timer::stop() noexcept
+{
+    if (*(m_waitSemaphore.getValue()) == static_cast<int>(posix::SemaphoreWaitState::TIMEOUT))
+    {
+        auto stopResult = m_waitSemaphore.post();
+        cxx::Expects(!stopResult.has_error());
```

Comment:
`Ensures` and `Expects` have exactly the same implementation. However, `Expects` is [preferred](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Ri-ensures) for pre-conditions and `Ensures` for post-conditions.

Reply:
Okay, Replaced with Ensures.